examples/pypi_wheel_provider.py

```diff
@@ -123,19 +123,20 @@ def get_base_requirement(self, candidate):
     def get_preference(self, resolution, candidates, information):
         return len(candidates)
 
-    def find_matches(self, requirement):
+    def iter_matches(self, requirements):
         candidates = []
-        name = requirement.name
-        extras = requirement.extras
+        name = requirements[0].name
+        extras = requirements[0].extras
```

Comment:
This will break if the various requirements have different extras. I'm not sure if that's possible (as the extras are included in the name) but again it's a complexity that should be called out in a comment, at least.

Reply:
The extras will always match for pip, but may not in this example. I’m not entirely sure whether it’s worth it to implement something as elaborated here though; maybe I’ll rewrite the examples into one after finishing this PR.

Reply:
Works for me. In general, though, I do think this might make extras more complex to handle in the general case, and having a worked example somewhere would be good.